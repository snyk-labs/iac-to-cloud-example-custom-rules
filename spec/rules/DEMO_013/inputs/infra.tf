resource "github_repository" "valid_by_node_id" {
  name = "valid-by-node-id"
}

resource "github_branch_protection" "valid_by_node_id" {
  repository_id = github_repository.valid_by_node_id.node_id
  pattern       = "main"
  push_restrictions = [
    "snyk/cloud-engines",
  ]
}

resource "github_repository" "valid_by_name" {
  name = "valid-by-name"
}

resource "github_branch_protection" "valid_by_name" {
  repository_id = github_repository.valid_by_name.name
  pattern       = "main"
  push_restrictions = [
    "snyk/cloud-engines",
  ]
}

resource "github_repository" "invalid" {
  name = "invalid"
}

resource "github_repository" "invalid_empty_restrictions" {
  name = "invalid-empty-restrictions"
}

resource "github_branch_protection" "invalid_empty_restrictions" {
  repository_id     = github_repository.invalid_empty_restrictions.name
  pattern           = "main"
  push_restrictions = []
}
