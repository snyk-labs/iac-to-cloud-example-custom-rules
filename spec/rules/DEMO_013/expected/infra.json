[
  {
    "passed": false,
    "ignored": false,
    "resource_id": "github_repository.invalid",
    "resource_namespace": "spec/rules/DEMO_013/inputs/infra.tf",
    "resource_type": "github_repository",
    "severity": "low",
    "resources": [
      {
        "id": "github_repository.invalid",
        "type": "github_repository",
        "namespace": "spec/rules/DEMO_013/inputs/infra.tf",
        "location": [
          {
            "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
            "line": 25,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "name"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
              "line": 26,
              "column": 3
            }
          },
          {
            "path": [
              "node_id"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
              "line": 25,
              "column": 1
            }
          }
        ]
      }
    ]
  },
  {
    "passed": false,
    "ignored": false,
    "resource_id": "github_repository.invalid_empty_restrictions",
    "resource_namespace": "spec/rules/DEMO_013/inputs/infra.tf",
    "resource_type": "github_repository",
    "severity": "low",
    "resources": [
      {
        "id": "github_branch_protection.invalid_empty_restrictions",
        "type": "github_branch_protection",
        "namespace": "spec/rules/DEMO_013/inputs/infra.tf",
        "location": [
          {
            "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
            "line": 33,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "push_restrictions"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
              "line": 36,
              "column": 3
            }
          },
          {
            "path": [
              "repository_id"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
              "line": 34,
              "column": 3
            }
          }
        ]
      },
      {
        "id": "github_repository.invalid_empty_restrictions",
        "type": "github_repository",
        "namespace": "spec/rules/DEMO_013/inputs/infra.tf",
        "location": [
          {
            "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
            "line": 29,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "name"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
              "line": 30,
              "column": 3
            }
          },
          {
            "path": [
              "node_id"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
              "line": 29,
              "column": 1
            }
          }
        ]
      }
    ]
  },
  {
    "passed": true,
    "ignored": false,
    "resource_id": "github_repository.valid_by_name",
    "resource_namespace": "spec/rules/DEMO_013/inputs/infra.tf",
    "resource_type": "github_repository",
    "severity": "low",
    "resources": [
      {
        "id": "github_branch_protection.valid_by_name",
        "type": "github_branch_protection",
        "namespace": "spec/rules/DEMO_013/inputs/infra.tf",
        "location": [
          {
            "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
            "line": 17,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "push_restrictions"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
              "line": 20,
              "column": 3
            }
          },
          {
            "path": [
              "repository_id"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
              "line": 18,
              "column": 3
            }
          }
        ]
      },
      {
        "id": "github_repository.valid_by_name",
        "type": "github_repository",
        "namespace": "spec/rules/DEMO_013/inputs/infra.tf",
        "location": [
          {
            "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
            "line": 13,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "name"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
              "line": 14,
              "column": 3
            }
          },
          {
            "path": [
              "node_id"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
              "line": 13,
              "column": 1
            }
          }
        ]
      }
    ]
  },
  {
    "passed": true,
    "ignored": false,
    "resource_id": "github_repository.valid_by_node_id",
    "resource_namespace": "spec/rules/DEMO_013/inputs/infra.tf",
    "resource_type": "github_repository",
    "severity": "low",
    "resources": [
      {
        "id": "github_branch_protection.valid_by_node_id",
        "type": "github_branch_protection",
        "namespace": "spec/rules/DEMO_013/inputs/infra.tf",
        "location": [
          {
            "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
            "line": 5,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "push_restrictions"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
              "line": 8,
              "column": 3
            }
          },
          {
            "path": [
              "repository_id"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
              "line": 6,
              "column": 3
            }
          }
        ]
      },
      {
        "id": "github_repository.valid_by_node_id",
        "type": "github_repository",
        "namespace": "spec/rules/DEMO_013/inputs/infra.tf",
        "location": [
          {
            "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
            "line": 1,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "name"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
              "line": 2,
              "column": 3
            }
          },
          {
            "path": [
              "node_id"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_013/inputs/infra.tf",
              "line": 1,
              "column": 1
            }
          }
        ]
      }
    ]
  }
]